drivers/ethernet/eth_enc28j60.c

```diff
@@ -360,6 +493,16 @@ static int eth_enc28j60_init(struct device *dev)
 		return -EINVAL;
 	}
 
+#ifndef CONFIG_SPI_LEGACY_API
+	/* Initialize GPIO */
+	if (gpio_pin_configure(enc_spi_cs.gpio_dev, config->spi_cs_pin,
+			       (GPIO_DIR_OUT | GPIO_PUD_PULL_DOWN))) {
+		SYS_LOG_ERR("Unable to configure GPIO pin %u",
+			    config->gpio_pin);
+		return -EINVAL;
+	}
+#endif
+
```

Comment:
[HOW COME github let me delete your comment, by mistake? how smart is this... sorry ]

spi_context_cs_configure() should do it, on spi driver side.

Could it be your board needs something specific? I see this GPIO_PUD_PULL_DOWN, looks like it's specific. (it's not supported by all gpio controller)

I could make a quick patch so extra gpio directives can be passed to gpio_pin_configure() in drivers/spi/spi_context.h

Reply:
no problem :-) , i might move that part into my board folder, thanks